src/Microsoft.AspNet.OData.Shared/Routing/Conventions/AttributeRoutingConvention.cs

```diff
@@ -59,6 +59,9 @@ private AttributeRoutingConvention(string routeName)
                 if (action.IsHttpMethodSupported(request.GetRequestMethodOrPreflightMethod()) && template.TryMatch(odataPath, values))
                 {
                     values["action"] = action.ActionName;
+                    // store the method info to allow the action selector to distinguish
+                    // between this action and other overloads
+                    values[ODataRouteConstants.MethodInfo] = action.GetMethodInfo();
```

Comment:
What if instead of storing an entire `MethodInfo` object, you stored a boolean value with a value of true if the method is decorated with `ODataRouteAttribute`; otherwise false?
I'm also curious about the existing method `public IEnumerable<T> GetCustomAttributes<T>(bool inherit) where T : Attribute`. Does it return the attributes on the action method?
If the existing `GetCustomAttribute` method can't do the job, you could add a method or property in the `WebApiActionDescriptor` class that returns true if the action method contains `ODataRouteAttribute`

Reply:
@gathogojr you can tell whether a method is decorated with `ODataRouteAttribute` by using the `actionDescriptor.GetCustomAttributes` method. The problem is that just relying on the `ODataRoute` attribute is not sufficient. I actually tried that first, and it causes tests to fail. The problem is that you can have two different overloads with `ODataRouteAttribute`, e.g. `Get()` and `Get(int key)`, or realistic, actions handling OData functions with the same name but different parameters. In this case the action selector would need to match parameters to know which action to select. But because the attribute routing has already done the heavy lifting, we could just use the exact method that it narrowed down to, which is what I was trying to achieve by passing the method info from the `AttributeRoutingConvention` to the action selector.

Reply:
Just to be clear, I'm not storing the `MethodInfo` so that I can access its attributes, I'm storing the `MethodInfo` so that I can compare it directly to method info instances in the action selector to find a match. I guess the alternative would to compare two parameters, to check whether two methods have the same sequence of parameter types.